Added other feature importances in python package

alistairewj opened the pull request:
From what I could tell the python package only implemented feature importances using `get_fscore()` which returned the number of times a feature was used to split data (I called this "weight", it was called "weight" in the R package). I added a function to calculate the average gain/coverage called `get_score()` with input importance_type. I also added it to plotting.py so the user can plot gain/coverage with `plot_importance` and a unit test to make sure I didn't botch anything.

terrytangyuan commented:
@alistairewj Thanks! Could you rebase and merge the unit tests with the other `feature_importance` tests that I merged earlier?

alistairewj commented:
No problem - I've made some updates though, which I'd like your opinion on. I changed the sklearn `feature_importances_` to return an Nx3 array, rather than an array of length N. The R package returns an Nx4 data table (first column are feature names) so I thought this was (reasonably) consistent. Also, `feature_importances_` returned the feature counts after being normalized, but I think that makes it inconsistent with the lower level `get_fscore()` so I removed that. I updated the tests to reflect this. Let me know what you think.

terrytangyuan commented:
New options are good but let's keep it backward compatible since this project has been integrated into many systems. If there's no bug or anything, there's no need to make those additional changes.

alistairewj commented:
Fair enough, I reverted that change.

terrytangyuan commented:
Thanks. Merged!
